### DISCUSSION TITLE
record diskUUID doesn't match the one on the disk

### DISCUSSION THREAD

**Discussion by btdan** [longhorn/longhorn#4694] [Q&A]
There are two longhorn node is not ready. And the reason is DiskFilesystemChanged. Just as the picture as follows:
![41b53282284400f7ca108a024a8ac6a](https://user-images.githubusercontent.com/5276480/194750266-eaa786b9-7533-4cb2-a6b4-a2574eaac44d.jpg)
It seems that the ssd disk is changed? But I do nothing for it.
How can I solve this problem?
Thanks very much.

**Comment by derekbit** [comment]
Can you check if the disk is somehow unmounted?

**Reply by btdan** [reply]
Thanks very much. I just remount this disk and format it before it is used.
And still now, it works well.
Thank you for your reply.

**Comment by Athosone** [comment]
It keeps happening for me too
My disk is mounted through proxmox to a microOS VM.
It keep swapping from two uuid. I wonder how it choses it...
Also when replacing the uuid the data is gone :(

**Reply by derekbit** [reply]
Does it happen after reboot?

**Reply by Athosone** [reply]
Yes, I think I understood why finally.
The fstab instructions mounted disks by referencing /dev/sda to /data/… instead of using the uuid of the disk.
Hence on reboot it might have swapped their mount point location sometimes.

does it make sense ?

**Comment by Joao-1** [comment]
Hi. I'm not sure why, but after a reboot my drive was renamed (from `/dev/sda` to `/dev/sdb`). As a result, the mounting instructions in `/etc/fstab` were wrong. I changed it to `dev/sdb` and ran `sudo mount -a`. This fixed the problem.

**Reply by Joao-1** [reply]
Now I know that the system can rename a disc after a reboot, so it is important to use the disc's UUID instead of the name.

**Comment by hellt** [comment]
how to deal with this error and Talos, where you can't ssh into the node and remount?

**Reply by hellt** [reply]
ok, using the longhorn UI I removed the unschedulable disk using the trashcan icon and then re-added it using the same /var/lib/longhorn path
![CleanShot 2025-07-06 at 15 33 21](https://github.com/user-attachments/assets/dde4277f-e49a-40d7-a849-2a98baa3bf20)

**Reply by RazeLighter777** [reply]
Can confirm this fixed it for me, though I needed to disable scheduling first, and salvage my nodes. I did this by manually salvaging each node, and then when that got too tedious, setting longhorn to automatically attempt to salvage nodes in the settings.
